src/main/kotlin/dev/arbjerg/ukulele/audio/Player.kt

```diff
@@ -101,7 +103,15 @@ class Player(val beans: Beans, guildProperties: GuildProperties) : AudioEventAda
         player.stopTrack()
     }
 
+    fun toggleRepeat() {
+        repeatOne = !repeatOne
+    }
+
     override fun onTrackEnd(player: AudioPlayer, track: AudioTrack, endReason: AudioTrackEndReason) {
+        if (repeatOne) {
+            queue.push(track.makeClone())
+        }
```

Comment:
If Repeat One is enabled, push a clone of the track that just finished onto the top of the stack/queue so that it plays again.